File: Test/StringShaperTest.java
package test;

import static org.junit.Assert.*;

import org.junit.Test;

import com.lucciola.haskell.HaskellAnywhere;
import com.lucciola.result.CommunicationResult;

public class StringShaperTest {

	@Test
	public void test() {
		HaskellAnywhere shaper = new HaskellAnywhere();
		CommunicationResult result = shaper.input("\"Hello, world\"");
		
		assertEquals(result.getMessage(), "\\\"Hello, world\\\"\\n");
	}

}

File: src/test/kotlin/StringShaperTest.kt
package test

import org.junit.Assert.*

import org.junit.Test

import com.lucciola.haskell.HaskellAnywhere
import com.lucciola.result.CommunicationResult

class StringShaperTest {
    @Test
    fun test() {
        val shaper = HaskellAnywhere()
        val result = shaper.input("\"Hello, world\"")

        assertEquals("\"Hello, world\"\n", result.message)
    }
}
